Mathlib/Tactic/NormNum/Core.lean

```diff
@@ -120,35 +120,28 @@ required in each use of a number literal at type `α`.
 -/
 @[simp] def Rat.rawCast [DivisionRing α] (n : ℤ) (d : ℕ) : α := n / d
 
-section
-set_option warningAsError false -- FIXME: prove the sorries
-
 theorem IsRat.to_isNat {α} [Ring α] : ∀ {a : α} {n}, IsRat a (.ofNat n) (nat_lit 1) → IsNat a n
-  | _, _, ⟨inv, eq⟩ => sorry
+  | _, _, ⟨inv, rfl⟩ => have := @invertibleOne α _; ⟨by simp⟩
 
 theorem IsNat.to_isRat {α} [Ring α] : ∀ {a : α} {n}, IsNat a n → IsRat a (.ofNat n) (nat_lit 1)
-  | _, _, ⟨rfl⟩ => sorry
+  | _, _, ⟨rfl⟩ => ⟨⟨1, by simp, by simp⟩, by simp⟩
 
 theorem IsRat.to_isInt {α} [Ring α] : ∀ {a : α} {n}, IsRat a n (nat_lit 1) → IsInt a n
-  | _, _, ⟨inv, eq⟩ => sorry
+  | _, _, ⟨inv, rfl⟩ => have := @invertibleOne α _; ⟨by simp⟩
 
 theorem IsInt.to_isRat {α} [Ring α] : ∀ {a : α} {n}, IsInt a n → IsRat a n (nat_lit 1)
-  | _, _, ⟨rfl⟩ => sorry
+  | _, _, ⟨rfl⟩ => ⟨⟨1, by simp, by simp⟩, by simp⟩
 
-theorem IsRat.to_raw_eq [DivisionRing α] : IsRat (a : α) n d → a = Rat.rawCast n d
-  | ⟨inv, eq⟩ => sorry
-
-theorem IsRat.of_raw (α) [DivisionRing α] (n : ℤ) (d : ℕ) : IsRat (Rat.rawCast n d : α) n d := sorry
```

Comment:
This theorem was false since `d` might not be invertible in `α`.

Reply:
we turned out not to need them anyway, so I'm fine with just removing them. (They might come back when we do rat-dependent `ring` functionality.) The fix is of course to assume that d is not zero when necessary, which will always be true in a normal form expression.

I am surprised the first one isn't true though. In that one we have an `IsRat` and hence an assumption that `d` is nonzero (since IIRC 0 is never invertible in a division ring because it assumes 0 != 1).